### ISSUE TITLE
Incorrect protocol check (appeared in the version 1.1.333)

### ISSUE THREAD

**Issue by qornev** [microsoft/pyright#6230]
Python built-in lists does not fit by type in numpy functions

```python
import numpy as np

arr: list[float] = [0.1, 0.2, 0.3]

np.random.choice(arr)
```

With libs

```
pyright==1.1.333
numpy==1.23.5
```

When try `pyright test.py` gives

```
/Users/qornev/workspace/test.py
  /Users/qornev/workspace/test.py:5:1 - error: No overloads for "choice" match the provided arguments (reportGeneralTypeIssues)
  /Users/qornev/workspace/test.py:5:18 - error: Argument of type "list[float]" cannot be assigned to parameter "a" of type "ArrayLike" in function "choice"
    Type "list[float]" cannot be assigned to type "ArrayLike"
      "list[float]" is incompatible with protocol "_SupportsArray[dtype[Unknown]]"
        "__array__" is not present
      "list[float]" is incompatible with protocol "_NestedSequence[_SupportsArray[dtype[Unknown]]]"
        "__getitem__" is an incompatible type
          One or more overloads of "__getitem__" is not assignable
            No overloaded function matches type "(index: int, /) -> (_T_co@_NestedSequence | _NestedSequence[_T_co@_NestedSequence])"
        "__iter__" is an incompatible type
    ... (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations 
```

It looks like every protocol from Union is being checked instead of just one for numpy [ArrayLike](https://github.com/numpy/numpy/blob/382eedf3891d474196677e65f3aae066afb32c5f/numpy/_typing/_array_like.py#L74) type

**Comment by erictraut** [comment]
I think pyright's new behavior is correct. You're using a very outdated version of numpy. If you upgrade to a more recent version, the error goes away.
